app/helpers/proxmox_vm_cloudinit_helper.rb

```diff
@@ -41,4 +41,49 @@ def parse_server_cloudinit(args)
     end
     cloudinit_h
   end
+
+  def create_cloudinit_iso(vm_name, configs)
+    iso = File.join('/var/lib/vz/template/iso', "#{vm_name}_cloudinit.iso")
+    arguments = ["-output #{iso}", '-volid', 'cidata', '-joliet', '-rock']
+    ssh = Fog::SSH.new(URI.parse(fog_credentials[:proxmox_url]).host, fog_credentials[:proxmox_username].split('@')[0], { password: fog_credentials[:proxmox_password] })
+    wd = ssh.run([%({mktemp -d})]).first.stdout.chomp
+    configs.each do |config|
+      begin
+        ssh.run([%({echo "#{config[1]}" >> "#{wd}/#{config[0]}"})])
+      rescue ::Foreman::Exception => e
+        logger.warn("Error writing to the file #{config[0]}: #{e}")
+      end
+      arguments.append(File.join(wd, config[0]))
+    end
+    logger.warn("iso image generation args: #{arguments}")
+    raise ::Foreman::Exception, _('ISO build failed') unless ssh.run([%({genisoimage #{arguments.join(' ')}})])
+    ssh.run([%(rm -rf #{wd})])
+    iso
+  end
+
+  def parse_cloudinit_config(args)
+    filenames = ["meta-data"]
+    config_data = ["instance-id: #{args[:name]}"]
+    if args[:user_data].include?('#network-config') && args[:user_data].include?('#cloud-config')
+      config_data.concat(args.delete(:user_data).split('#network-config'))
+      filenames.append(['user-data', 'network-config'])
+    elsif args[:user_data].include?('#network-config') && !args[:user_data].include?('#cloud-config')
+      config_data.append(args.delete(:user_data).split('#network-config')[1])
+      filenames.append("network-config")
+    elsif !args[:user_data].include?('#network-config') && args[:user_data].include?('#cloud-config')
+      config_data.append(args.delete(:user_data))
+      filenames.append("user-data")
+    end
+
+    configs = filenames.zip(config_data).to_h
+
+    iso = create_cloudinit_iso(args[:name], configs)
```

Comment:
we should not start the cloudinit process,if its not a cloud-init config

Reply:
Fixed